src/Avalonia.Controls/Control.cs

```diff
@@ -472,14 +473,32 @@ internal AutomationPeer GetOrCreateAutomationPeer()
             return _automationPeer;
         }
 
+        /// <inheritdoc/>
+        protected override void OnPointerPressed(PointerPressedEventArgs e)
+        {
+            base.OnPointerPressed(e);
+
+            var contextGestures = AvaloniaLocator.Current.GetService<PlatformPointerConfiguration>()?.OpenContextMenu;
+
+            if (e.Source == this
+                && !e.Handled
+                && contextGestures?.Any(x => x.Matches(e)) == true)
+            {
+                e.Handled = true; // Prevent click if it is context menu event
+            }
+        }
+
         /// <inheritdoc/>
         protected override void OnPointerReleased(PointerReleasedEventArgs e)
         {
             base.OnPointerReleased(e);
 
+            var contextGestures = AvaloniaLocator.Current.GetService<PlatformPointerConfiguration>()?.OpenContextMenu;
+
             if (e.Source == this
                 && !e.Handled
-                && e.InitialPressMouseButton == MouseButton.Right)
+                && e.InitialPressMouseButton == MouseButton.Right
```

Comment:
Shouldn't the button be encoded in the gesture?
I'm also not sure that we should change the PointerPressed behavior.

@maxkatz6 @grokys

Reply:
also test hte caste of when they swtich left and right hand mouse..  in windows ts aways a problems.
 Open System Preferences on Mac. In System Preferences, locate "Mouse" right /left side

Reply:
Removed InitialPressMouseButton condition, I mistakenly forgot it.

About left-handed mice. This is a good point. current PR does not consider such case. To detect mouse configuration its needed to implement reading platform-specific system settings. I think it is out of scope of this PR, but I can try to investigate this later.